tests/test_unit/test_move_accessor.py

```diff
@@ -1,42 +1,120 @@
+
+
+@pytest.mark.parametrize(
+    "invalid_dataset",
+    (
+        "not_a_dataset",
+        "empty_dataset",
+        "missing_var_poses_dataset",
+        "missing_var_bboxes_dataset",
+        "missing_dim_poses_dataset",
+        "missing_dim_bboxes_dataset",
+    ),
+)
+def test_compute_kinematics_with_invalid_dataset(
+    invalid_dataset, kinematic_property, request
+):
+    """Test that computing a kinematic property of an invalid
+    poses or bounding boxes dataset via accessor methods raises
+    the appropriate error.
+    """
+    invalid_dataset = request.getfixturevalue(invalid_dataset)
+    expected_exception = (
+        RuntimeError
+        if isinstance(invalid_dataset, xr.Dataset)
+        else AttributeError
     )
-    def test_invalid_method_call(self, valid_poses_dataset, method):
-        """Test that invalid accessor method calls raise an AttributeError."""
-        with pytest.raises(AttributeError):
-            getattr(valid_poses_dataset.move, method)()
+    with pytest.raises(expected_exception):
+        getattr(invalid_dataset.move, f"compute_{kinematic_property}")()
+
+
+@pytest.mark.parametrize(
+    "method", ["compute_invalid_property", "do_something"]
+)
+@pytest.mark.parametrize(
+    "valid_dataset", ("valid_poses_dataset", "valid_bboxes_dataset")
+)
+def test_invalid_move_method_call(valid_dataset, method, request):
+    """Test that invalid accessor method calls raise an AttributeError."""
+    valid_input_dataset = request.getfixturevalue(valid_dataset)
+    with pytest.raises(AttributeError):
+        getattr(valid_input_dataset.move, method)()
+
+
+@pytest.mark.parametrize(
+    "input_dataset, expected_exception, log_message",
+    (
+        (
+            "valid_poses_dataset",
+            does_not_raise(),
+            "",
+        ),
+        (
+            "valid_bboxes_dataset",
+            does_not_raise(),
+            "",
+        ),
+        (
+            "valid_bboxes_dataset_in_seconds",
+            does_not_raise(),
+            "",
+        ),
+        (
+            "missing_dim_poses_dataset",
+            pytest.raises(ValueError),
+            "The dataset does not contain valid poses. "
+            "Missing required dimensions: ['time']",
+        ),
+        (
+            "missing_dim_bboxes_dataset",
+            pytest.raises(ValueError),
+            "The dataset does not contain valid bboxes. "
+            "Missing required dimensions: ['time']",
+        ),
+        (
+            "missing_var_poses_dataset",
+            pytest.raises(ValueError),
+            "The dataset does not contain valid poses. "
+            "Missing required data variables: ['position']",
+        ),
+        (
+            "missing_var_bboxes_dataset",
+            pytest.raises(ValueError),
+            "The dataset does not contain valid bboxes. "
+            "Missing required data variables: ['position']",
+        ),
+    ),
+)
+def test_move_validate(
```

Comment:
I see that this test is added just because we never use "validate" with bboxes. With poses data, we also only call this once in `save_poses`. Maybe we should open an issue to discuss this?

Reply:
Added #257 - feel free to add anything you feel I didn't cover.

I will add it to the behaviour meeting agenda for next time, I remember a similar conversation happening a while ago.